### PR TITLE
Add Rename collection

### PR DESCRIPTION
**Pull request by NicolasGB**
This PR contains the same code as Wolfx's but his fork repo has been deleted.

Please consider to review this feature as it's still relevant and very useful.

I can make the same changes for the v2 driver if required.

### CODE REVIEW

File: collection.go [hunk 1]
```diff
@@ -77,6 +77,10 @@ type Collection interface {
 	// Truncate removes all documents from the collection, but leaves the indexes intact.
 	Truncate(ctx context.Context) error
 
+	// Rename renames the collection.
+	// If the collection does not exist, a NotFoundError is returned.
+	Rename(ctx context.Context, options SetRenameOptions) error
```
Comment (nikita-vanyasin) [inline collection.go:82]: Although passing an options object here is obviously a better solution than passing only a newName, we should stick with existing style. 

Please change it simple `Rename(ctx context.Context, newName string) error`.
See `View` interface which already has Rename function.

File: collection.go [hunk 2]
```diff
@@ -77,6 +77,10 @@ type Collection interface {
 	// Truncate removes all documents from the collection, but leaves the indexes intact.
 	Truncate(ctx context.Context) error
 
+	// Rename renames the collection.
```
Comment (nikita-vanyasin) [inline collection.go:80]:
```suggestion
	// Rename renames the collection (SINGLE server only)..
```

**Review by nikita-vanyasin — CHANGES_REQUESTED** [review]
Hello @NicolasGB 

Thank you for contribution!
We can merge this PR with next adjustments:
1) Please add test for Rename function. Ensure that test is not executed in Cluster setup (rename feature is not supported there). (you can use `skipNoSingle` helper for test).
2) see code comments / suggestions

File: collection.go (hunk 1 above)
Comment (NicolasGB) [reply collection.go]: Thanks for your review. 
I've adapted the changes, now the interface function matches the View interface.

I am having some trouble running my tests locally, all of them return "No endpoints found", is there any guide for setting up the test environment locally?

**Comment by fceller** [comment]
CLA available

File: collection.go (hunk 1 above)
Comment (NicolasGB) [reply collection.go]: I followed MAINTAINERS.md.

I ran all tests and seem to be passing locally, is it possible to manually trigger the CI or do i need to make another commit?

**Review by nikita-vanyasin — APPROVED** [review]

**Review by jwierzbo — APPROVED** [review]

**Comment by nikita-vanyasin** [comment]
@NicolasGB thank you!
Auto-checks are executed only on verified commits.
To run this checks, please do next steps:
- configure commits [signing](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits) for your local repo (`git config commit.gpgsign true`)
- squash and rebase all commits in your branch and force-push again
- that should run the checks automatically, 

If you now squash your commits and force-push the branch again, the check will be able to start automatically - they are not executed until you have CLA-signed commits

**Comment by NicolasGB** [comment]
No problem! Thanks for guiding me through this PR!

**Comment by nikita-vanyasin** [comment]
@NicolasGB now just one thing is missing: can you please add a line to CHANGELOG file about your change? This file is maintained manually for now, unfortunately.

**Comment by NicolasGB** [comment]
@nikita-vanyasin  I've added the Changelog message, i tried to follow the convention, let me know if the message is suitable enough or it needs improvements

**Pull request merged**